trio/_core/tests/test_thread_cache.py

```diff
@@ -25,6 +25,29 @@ def deliver(outcome):
         outcome.unwrap()
 
 
+def test_thread_cache_deref():
+    res = [False]
+
+    class del_me:
+        def __call__(self):
+            return 42
+
+        def __del__(self):
+            res[0] = True
```

Comment:
so this is the instance's destructor, where my original report was on the class destructor

but I see that the bug indeed affects both of these.  (I suppose that not destroying the instance is enough to keep a reference to the class?)

does this variant of the test fail before the patch?

Reply:
> does this variant of the test fail before the patch?

I see it does 👍

Reply:
I couldn't easily reproduce the problem using a class destructor, so I opted for a minimal test that shows the reference to be gone / some destructor to be called.